### PR TITLE
Add a basic sitemap generator

### PR DESCRIPTION
**Pull request by peteryates**
### Trello card

https://trello.com/c/CEFa8IGk/1401-create-or-update-our-sitemap

### Context and changes

It uses both the combined frontmatter from all content (via `Pages::Frontmatter`) and a hard-coded list of extra paths, which aren't generated from markdown files (like /events) to build a [sitemap](https://www.sitemaps.org/protocol.html).

### Guidance to review

Find it at `/sitemap.xml`

### CODE REVIEW

File: spec/requests/sitemap_controller_spec.rb [hunk 1]
```diff
@@ -0,0 +1,76 @@
...
+      },
+      "/my-story-into-teaching/what-a-great-story" => {
+        "name" => "What a great story",
+        "priority" => 0.7,
+        "date" => "2020-10-10",
+      },
+    }
+  end
+
+  before { allow(Pages::Frontmatter).to receive(:list) { nodes } }
```
Comment (ethax-ross) [inline spec/requests/sitemap_controller_spec.rb:25]: It took me a second to realise what `nodes` were... would `pages` be clearer?

File: app/controllers/sitemap_controller.rb [hunk 2]
```diff
@@ -0,0 +1,46 @@
...
+
+  ALIASES = {
+    "/home" => "/",
+  }.freeze
+
+  OTHER_PATHS = %w[
+    /events
+    /event-categories/train-to-teach-events
+    /event-categories/online-q-as
+    /event-categories/school-and-university-events
```
Comment (ethax-ross) [inline app/controllers/sitemap_controller.rb:12]: Should the privacy policy/cookies/guidance pages be in here? and would it be worth indexing the sign up for the mailing list initial page?

File: app/controllers/sitemap_controller.rb [hunk 3]
```diff
@@ -0,0 +1,46 @@
+class SitemapController < ApplicationController
+  DEFAULT_LASTMOD = "2021-03-01".freeze
```
Comment (ethax-ross) [inline app/controllers/sitemap_controller.rb:2]: Is there a particular rational behind this date, or is it just a case of it needs to be static and in the past? I guess it makes sense as a sort of 'when the website launched' date?

**Review by ethax-ross — APPROVED** [review]
Looks good, couple of observations but they are take it or leave it

File: app/controllers/sitemap_controller.rb (hunk 3 above)
Comment (peteryates) [reply app/controllers/sitemap_controller.rb:2]: Yeah, I'm planning to do a sweep of the content repo and add `priority` and `date` entries to the markdown, I just picked a past date here for the time being.

File: spec/requests/sitemap_controller_spec.rb (hunk 1 above)
Comment (peteryates) [reply spec/requests/sitemap_controller_spec.rb:25]: Ah yeah, good call 👍🏽

**Review by ethax-ross — APPROVED** [review]

**Pull request merged**